chore: Add deprecation notes to customStateCheck (#1708)
kyma-project/lifecycle-manager: docs/technical-reference/api/moduleTemplate-cr.md

### **.spec.customStateCheck**

> **Warning** This field is deprecated and will be removed in the future. 

Reviewer: ```suggestion
> **CAUTION:** This field is deprecated and will be removed in the future. 
```

> > **CAUTION:** This field is deprecated and will be removed in the future. 

Reviewer: Let's provide the deprecation date also here.

> > **CAUTION:** This field is deprecated and will be removed in the future. 

Reviewer: ```suggestion
> **CAUTION:** This field is deprecated and will be removed on {DATE}. 
```

> > **CAUTION:** This field is deprecated and will be removed in the future. 

Reviewer: ```suggestion
> **CAUTION:** This field was deprecated at the end of July 2024 and will be removed on {DATE}. 
```
We could also provide the deletion date. Is there any action required from the users?
